### PR TITLE
Fix/205 recent activities refactor

### LINKED ISSUE #205: Recent Activities Refactor
**Describe the task**
Refactor the component and fix minor issues. The issues were listed in JIRA task [#474](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-474)

Issues
- Change mouse icon to clickable when hover "view icon" on "My recent activities - request"
- Update date format on "My recent activities - recents." table
- Centralize icon on View column "My recent activities - recents."
- Missing icon on Activity type "My recent activities - requests" 
- Update tip message to view icon on "My recent activities - recents.", validate the correct message, because on mock and hand-off notes the message is different
- Scroll doesn't works to side on "My recent activities - recents." when screen is medium size
- Update tag name on column status on "My recent activities - recents.", validate the correct name, because on mock and hand-off notes the names are different

**Acceptance Criteria**
- [ ] Component refactored to current code standards
- [ ] All minor issues fixed

### PR DESCRIPTION
**Pull request by mgaseta**
# Description

Fixed all small issues and refactored the component.

- [x] Change mouse icon to clickable when hover "view icon" on "My recent activities - request"
- [x] Update date format on "My recent activities - recents." table
- [x] Centralize icon on View column "My recent activities - recents."
- [x] Missing icon on Activity type "My recent activities - requests"
- [x] Update tip message to view icon on "My recent activities - recents.", validate the correct message, because on mock and hand-off notes the message is different
- [x] Scroll doesn't works to side on "My recent activities - recents." when screen is medium size
- [ ] Update tag name on column status on "My recent activities - recents.", validate the correct name, because on mock and hand-off notes the names are different
- This last one was not corrected, the status component will receive some new values that the UX team will define. So, to not extend this task anymore, this will be made in a separate one.
- The changes on the activity and files types are not permanent, we still need to discuss how those features will work with the backend.
- Some changes may seem off with the design, but those were agreed with the UX team, there are comments on the figma file explaining them.

Fixes #205 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Manual tests and unit tests.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-322-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-322-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-322-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-322-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-322-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-322-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

### CODE REVIEW

File: frontend/src/components/RecentActivities/RecentActivitiesTable/index.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                  hasIconOnly
+                  iconDescription={`View ${docTable ? 'file' : 'request'}`}
+                  tooltipPosition="bottom"
+                  kind="ghost"
+                  onClick={() => clickFn(item.id)}
+                  renderIcon={Icons.DataViewAlt}
+                  size="md"
+                />
+                {
+                  docTable
```
Comment (craigyu) [inline frontend/src/components/RecentActivities/RecentActivitiesTable/index.tsx:137]: Use boolean [naming convention](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig) on this variable.
`hasDocTable`?

**Review by craigyu — CHANGES_REQUESTED** [review]
👨🏻‍🔧 Great refactoring work here, I left a couple comments

File: frontend/src/components/RecentActivities/RecentActivitiesTable/index.tsx (hunk 1 above)
Comment (mgaseta) [reply frontend/src/components/RecentActivities/RecentActivitiesTable/index.tsx]: Decided to go with `isDocTable`, sounds more right, fix is [here](https://github.com/bcgov/nr-spar/pull/322/commits/5905ca83715ef2a5c29850f0f71a412368618dff)

File: frontend/src/components/RecentActivities/RecentActivitiesTable/index.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+  return (
+    <Table size="lg" className="activity-table">
+      <TableHead>
+        <TableRow>
+          {headers.map((header, idx) => (
+            <TableHeader
+              key={header}
+              id={`header-${header}-${idx}`}
+              data-testid={`header-${header}-${idx}`}
```
[the remark this answers is not in the document]
Comment (mgaseta) [reply frontend/src/components/RecentActivities/RecentActivitiesTable/index.tsx]: Fixed [here](https://github.com/bcgov/nr-spar/pull/322/commits/c5d866cd146ee87a3635937e816ba57d60f48c6a)

**Review by RMCampos — APPROVED** [review]
LGTM

**Comment by RMCampos** [comment]
@mgaseta @craigyu SonarCloud is failing due to low code coverage tests. Just highlighting. Besides that, it's all good. Great job 💯

**Comment by craigyu** [comment]
> @mgaseta @craigyu SonarCloud is failing due to low code coverage tests. Just highlighting. Besides that, it's all good. Great job 💯

We can't disable the 80% requirement so I'll have to manually override it until we use cypress for coverage

**Review by craigyu — APPROVED** [review]
Great work! especially on taking the initiative to refactor the two tables 👊

**Pull request merged**